src/rai_core/rai/tools/ros2/base.py

```diff
@@ -62,6 +62,60 @@ def is_writable(self, topic: str) -> bool:
             return True
         return topic in self.writable
 
+    def _check_permission_and_include(
+        self, name: str, check_readable: bool = True
+    ) -> Tuple[bool, bool, bool]:
+        """
+        Check permissions and determine if resource should be included.
+
+        Args:
+            name: Resource name (topic/service/action)
+            check_readable: If False, only checks writable (for services/actions).
+                           If True, checks both readable and writable (for topics).
```

Comment:
Minor formatting suggestion:
```suggestion
                            If True, checks both readable and writable (for topics).
```